enhancements/sdk-kubernetes-1.21.md

```diff
@@ -0,0 +1,219 @@
+---
+title: Support Kubernetes 1.21 in Operator SDK
+authors:
+  - "@varshaprasad96"
+reviewers:
+  - "@estroz"
+  - "@jmrodri"
+approvers:
+  - "@estroz"
+  - "@jmrodri"
+creation-date: 2021-06-10
+last-updated: 2020-06-10
+status: provisional
+see-also:
+  - "https://github.com/kubernetes/sig-release/tree/master/releases/release-1.21"
+replaces: []
+superseded-by: []
+---
+
+# Support Kubernetes 1.21 in Operator SDK
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] ~~Graduation criteria for dev preview, tech preview, GA~~ (N/A)
+
+## Open Questions
+
+
+## Summary
+
+Operator SDK consists of many components (project scaffolding, OLM integrations, scorecard, helm, ansible and java operators, and a Go library for operator authors) that have interactions with Kubernetes clusters. The goal of this enhancement is to upgrade ALL components of Operator SDK to be compatible with Kubernetes 1.21.
+
+## Motivation
+
+Operator SDK needs to support the latest features of Kubernetes since that operator developers expect to be able to use them shortly after the release is made.
+
+## Dependencies
+
+- [external] Require controller-runtime release that supports 1.21
+- [external] Require Helm release that supports 1.21
+- [external] Require kubebuilder release that scaffolds projects using 1.21
+- [internal] Require operator-framework/api release that supports 1.21
+- [internal] Require operator-framework/operator-lib release that supports 1.21
+
+### Goals
+
+- Operator SDK’s dependencies are upgraded to use Kubernetes 1.21.
+- All e2e tests run during Operator SDK CI are upgraded to test against Kubernetes 1.21.
+- New projects scaffolded with Operator SDK are upgraded to use Kubernetes 1.21 dependencies.
+- New projects scaffolded with Operator SDK should generate v1 CRDs by default.
```

Comment:
This is not necessarily a requirement since v1beta1 CRDs will be removed in k8s v1.22, but can be done now to prepare for that version's release. Make a note of this here in a sub-bullet.